bluesky_queueserver/manager/plan_queue_ops.py

```diff
@@ -0,0 +1,146 @@
+import aioredis
+import json
+
+
+class PlanQueueOperations:
+    def __init__(self, redis_host="localhost"):
+        self._redis_host = redis_host
+        self._name_running_plan = "running_plan"
+        self._name_plan_queue = "plan_queue"
+        self._r_pool = None
+
+    async def start(self):
+        self._r_pool = await aioredis.create_redis_pool(f"redis://{self._redis_host}", encoding="utf8")
+
+    async def delete_pool_entries(self):
+        """
+        Delete pool entries used by RE Manager. This method is mostly for use in testing.
+        """
+        await self._r_pool.delete(self._name_running_plan)
+        await self._r_pool.delete(self._name_plan_queue)
+
+    # -------------------------------------------------------------
+    #                       Running Plan
+
+    async def exists_running_plan_info(self):
+        """
+        Check if plan exists in the ppol
+        """
+        return await self._r_pool.exists(self._name_running_plan)
+
+    async def init_running_plan_info(self):
+        """
+        Initialize running plan info: create Redis entry that hold empty plan ({})
+        a record doesn't exist.
+        """
+        # Create entry 'running_plan' in the pool if it does not exist yet
+        if (not await self.exists_running_plan_info()) or (not await self.get_running_plan_info()):
+            await self.clear_running_plan_info()
+
+    async def set_running_plan_info(self, plan):
+        """
+        Write info on the currently running to Redis
+        """
+        await self._r_pool.set(self._name_running_plan, json.dumps(plan))
+
+    async def get_running_plan_info(self):
+        """
+        Read info on the currently running plan from Redis.
+
+        Returns
+        -------
+        dict
+            Dictionary representing currently running plan. Empty dictionary if
+            no plan is currently running.
+        """
+        return json.loads(await self._r_pool.get(self._name_running_plan))
```

Comment:
```suggestion
        plan = await self._r_pool.get(self._name_running_plan)
        return json.loads(plan) if plan else {}
```

We need to be defensive here as well.  Maybe premature, but this is something that bit me because for some reason I ended up with `''` in this key and I could not get the queueserver to start until I manually set `'{}'` to it.

Reply:
I am also :+1: on kicking that can down the road.

Reply:
Strange. I check if the key exists at startup and create it if it doesn't exist. But another layer of safety wouldn't hurt.

Reply:
I will put this change in the next PR. I think that classes like this should be 100% reliable and not cause problems. It allows to be more relaxed in development of the rest of the program.